### DISCUSSION TITLE
Building an app for Android using Rocket

### DISCUSSION THREAD

**Discussion by Raj2032** [rwf2/Rocket#2037] [General]
![image](https://user-images.githubusercontent.com/29293830/147403760-cf2fc624-f39a-4043-92be-1cc2d2d60b40.png)

Hey guys,

I am looking to make a web based application for Android which can scan qr codes, image process it using Rust.

The qr code would be a third party library I would use into my application, something like `rqrr`

https://github.com/WanzenBug/rqrr

If I compile the entire project including the `rqrr` library into a web application, would it just work out of the box or do i have to drastically change something?

There are two libraries I looked at which are Rocket UI:

https://rocket.rs

and Tauri:

https://tauri.studio/en/

Would they be able to acheive what I am looking for? And if so which one should I use in terms of ease of use, animations, and see the screenshot how there is a camera view in the middle of the screen and the rest is the app, would that be also supported?

**Comment by IniterWorker** [comment]
Hi @Raj2032,

Rocket.rs is not a UI lib. 

Your discussion is not related to the Rocket.rs framework.

B. r

**Reply by Raj2032** [reply]
sorry my mistake

**Reply by somehowchris** [reply]
As answered in https://github.com/SergioBenitez/Rocket/discussions/2034 and said by @IniterWorker. They do not pursue the same goals. Now you said that you would do it as a web-based application. Do you mean something like electron or tauri, building some website or really just receiving the image on a backend and pass it into the lib and return the result?
